modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java

```diff
@@ -1195,7 +1196,7 @@ else if (state.active()) {
         DiscoveryDataClusterState joiningNodeState;
 
         try {
-            joiningNodeState = marsh.unmarshal((byte[])discoData.joiningNodeData(), Thread.currentThread().getContextClassLoader());
+            joiningNodeState = marsh.unmarshal((byte[])discoData.joiningNodeData(), IgniteUtils.gridClassLoader());
```

Comment:
But this is not the classloader of Ignite configuration. This is the IgniteUtils.class.getClassloader().

1. Don't use IgniteUtils, but U.
2. Do something like U.resolveClassLoader(ctx.config());